include/nuttx/lib/lib.h

```diff
@@ -114,10 +114,12 @@ void lib_stream_release(FAR struct task_group_s *group);
 #ifdef CONFIG_STDIO_DISABLE_BUFFERING
 #  define lib_sem_initialize(s)
 #  define lib_take_semaphore(s)
+#  define lib_try_take_semaphore(s)
```

Comment:
how about rename lib_xxx_semaphore to fxxxlockfile directly?

Reply:
yes, maybe keeping the name and the indirection level is of no value and probably the code will become more clear to people familiar with the POSIX naming. I will try to rename it in an extra commit of this draft (this way if desired, it can be collapsed easily into one single commit before making it a non-draft PR).